build/source/dshare/get_ixname.f90

```diff
@@ -836,9 +818,135 @@ function get_ixbvar(varName)
+  ! get to here if cannot find the variable
+  case default
+   get_VarTypeName = 'missing'
+ endselect
+ end function get_VarTypeName
+
+ ! *******************************************************************************************************************
+ ! public subroutine get_ixUnknown: get the index of the named variable type from ANY structure, as well as the 
+ ! structrue that it was found in
+ ! *******************************************************************************************************************
+ subroutine get_ixUnknown(varName,typeName,vDex,err,message)
+ USE nrtype
+ USE globalData,only:structInfo        ! information on the data structures                  
+ USE multiconst,only:integerMissing    ! missing integer value
+ implicit none
+
+ ! dummies
+ character(*),intent(in)  :: varName   ! variable name
+ character(*),intent(out) :: typeName  ! variable type name
+ integer(i4b),intent(out) :: vDex      ! variable index in structure
+ integer(i4b),intent(out) :: err       ! error code
+ character(*),intent(out) :: message   ! error message
+
+ ! internals
+ integer(i4b)             :: iStruc    ! index for looping through structure types
+
+ ! error init
+ err=0
+ message='get_ixUnknown/'
+
+ ! loop through all structure types to find the one with the given variable name
+ ! pill variable index plus return which structure it was found in
+ do iStruc = 1,size(structInfo)
+  select case(trim(structInfo(iStruc)%structName))
+   case ('time' ); vDex = get_ixTime(trim(varName))
+   case ('forc' ); vDex = get_ixForce(trim(varName))
+   case ('attr' ); vDex = get_ixAttr(trim(varName))
+   case ('type' ); vDex = get_ixType(trim(varName))
+   case ('mpar' ); vDex = get_ixParam(trim(varName))
+   case ('indx' ); vDex = get_ixIndex(trim(varName))
+   case ('prog' ); vDex = get_ixProg(trim(varName))
+   case ('diag' ); vDex = get_ixDiag(trim(varName))
+   case ('flux' ); vDex = get_ixFlux(trim(varName))
+   case ('bpar' ); vDex = get_ixBpar(trim(varName))
+   case ('bvar' ); vDex = get_ixBvar(trim(varName))
+   case ('deriv'); vDex = get_ixDeriv(trim(varName))
+  endselect
+  if (vDex>0) then; typeName=trim(structInfo(iStruc)%structName); return; endif;
+ enddo
+
+ ! 404
+ err=20;message=trim(message)//'variable not found in any structure:'//trim(varName); return;
+
+ end subroutine get_ixUnknown
+
+ ! ***************************************************************************************************************
+ ! public function get_statName: get the name of the output statistics type
+ ! ***************************************************************************************************************
+ function get_statName(istat)
+ USE var_lookup,only:iLookStat                   ! indices of the possible output statistics
+ implicit none
+ ! define dummy variables
+ integer(i4b), intent(in) :: istat               ! stat type name
+ character(LEN=4)         :: get_statName        ! index of the named variable type list
```

Comment:
premature optimization - there is no good reason to limit this to 4 characters. F77 is a pain to read in part because "ppl flt cmplld t s nl lmtd nmbr f chrctrs" (people felt compelled to use only a limited number of characters). As we may want to add additional methods, it would be nice to just expand this.